packages/flutter/test/cupertino/button_test.dart

```diff
@@ -885,6 +885,66 @@ void main() {
     await tester.pump();
     expect(value, isTrue);
   });
+
+  testWidgets('Press and move on button and animation works', (WidgetTester tester) async {
+    await tester.pumpWidget(
+      boilerplate(child: CupertinoButton(onPressed: () {}, child: const Text('Tap me'))),
+    );
+    final TestGesture gesture = await tester.startGesture(
+      tester.getTopLeft(find.byType(CupertinoButton)),
+      kind: switch (defaultTargetPlatform) {
+        TargetPlatform.iOS ||
+        TargetPlatform.android ||
+        TargetPlatform.fuchsia => PointerDeviceKind.touch,
+        TargetPlatform.macOS ||
+        TargetPlatform.windows ||
+        TargetPlatform.linux => PointerDeviceKind.mouse,
+      },
+    );
+    addTearDown(gesture.removePointer);
+    // Check opacity
+    final FadeTransition opacity = tester.widget(
+      find.descendant(of: find.byType(CupertinoButton), matching: find.byType(FadeTransition)),
+    );
+    await tester.pumpAndSettle();
+    expect(opacity.opacity.value, 0.4);
+    final double moveDistance = CupertinoButton.tapMoveSlop();
+    await gesture.moveBy(Offset(0, -moveDistance + 1));
+    await tester.pumpAndSettle();
+    expect(opacity.opacity.value, 0.4);
+    await gesture.moveBy(const Offset(0, -2));
+    await tester.pumpAndSettle();
+    expect(opacity.opacity.value, 1.0);
+    await gesture.moveBy(const Offset(0, 1));
+    await tester.pumpAndSettle();
+    expect(opacity.opacity.value, 0.4);
+  }, variant: TargetPlatformVariant.all());
+
+  testWidgets(
+    'When lifting up outside the button slop after moving, onPressed should not be triggered.',
+    (WidgetTester tester) async {
+      bool value = false;
+      await tester.pumpWidget(
+        boilerplate(
+          child: CupertinoButton(
+            onPressed: () {
+              value = true;
+            },
+            child: const Text('Tap me'),
+          ),
+        ),
+      );
+      final TestGesture gesture = await tester.startGesture(
+        tester.getTopLeft(find.byType(CupertinoButton)),
+      );
+      await gesture.moveTo(
+        tester.getBottomRight(find.byType(CupertinoButton)) +
+            Offset(0, CupertinoButton.tapMoveSlop()),
+      );
+      await gesture.up();
+      expect(value, isFalse);
+    },
+  );
```

Comment:
Can you add a test:
*     'When lifting up inside the button slop after moving out then back, onPressed should be triggered.',

(Or you can append this check to the first test you added.)